### ISSUE TITLE
Implement Reacter & Reactant Facades

### ISSUE THREAD

**Issue by antonkomarev** [cybercog/laravel-love#54]
Since package API is strict typed its might be harder to use and couple your application with the package subsystems. Developers will pass to the `reactTo` method reactable instance and string value of reaction type name: `->reactTo($comment, 'Like')`

Facade design pattern to the rescue. Not the same as Laravel Facades (which act as "static proxies").

*These facades will replace experimental global `Cog\Laravel\Love\Facades\Love` facade.*

## Facade concept

`Facade` shields developers from the complex details of the system and provides them with a simplified view of it which is easy to use. It also decouples the code that uses the system from the details of the subsystems, making it easier to modify the system later. It decreases maintenance cost, but adds additional layer of code & decreases runtime performance.

## Reacter Facade

User via `Reacterable` trait will have additional method which will provide simpler access to `Reacter` model methods. This method returns `Facades\Reacter` class which hide internal complexity behind a single interface that appears simple from the outside. Method will have `via` (by way of) prefix.

```php
trait Reacterable
{
    public function viaLoveReacter(): ReacterFacade
    {
        return new ReacterFacade($this->getLoveReacter());
    }
}
```

### Usage

Instead default strict typed Reacter model calls:

```php
$reactant = $comment->getLoveReactant();
$reactionType = ReactionType::fromName('Like');

$user->getLoveReacter()->reactTo($reactant, $reactionType);
```

`Facades\Reacter` class has simpler interface:

```php
$user->viaLoveReacter()->reactTo($comment, 'Like');
```

Under the hood it will resolve Reactant model from `$comment` & `ReactionType` from the string name and call strict typed method.

## Reactant Facade

Comment via `Reactable` trait will have additional method which will provide simpler access to `Reactant` model methods. This method returns `Facades\Reactant` class which hide internal complexity behind a single interface that appears simple from the outside. Method will have `via` (by way of) prefix.

```php
trait Reactable
{
    public function viaLoveReactant(): ReactantFacade
    {
        return new ReactantFacade($this->getLoveReactant());
    }
}
```

### Usage

Instead default strict typed Reactant model calls:

```php
$reacter = $user->getLoveReacter();
$reactionType = ReactionType::fromName('Like');

$comment->getLoveReactant()->isReactedByWithType($reacter, $reactionType);
```

`Facades\Reactant` class has simpler interface:

```php
$comment->viaLoveReactant()->isReactedBy($user, 'Like');
```

Under the hood it will resolve Reacter model from `$user` & `ReactionType` from the string name and call strict typed method.

## Naming

`Facades\Reacter` & `Facades\Reactant` instantiation methods name candidate variants described in comments below.

**Comment by antonkomarev** [comment]
1. `actAs` prefix (too verbose)

```php
$user->actAsLoveReacter()->reactTo($article, 'Like');
```

**Comment by antonkomarev** [comment]
2. `as` prefix

```php
$user->asLoveReacter()->reactTo($article, 'Like');
```

Related reading: https://softwareengineering.stackexchange.com/questions/352157/whats-the-difference-between-to-and-as-method-name-prefixes

**Comment by antonkomarev** [comment]
3. `aka` prefix (use if `as` prefix not applicable)

```php
$user->akaLoveReacter()->reactTo($article, 'Like');
```

**Comment by antonkomarev** [comment]
### Name Variant №2

#### Convention

Method should have `as` prefix and `Facade` suffix.

```php
$user->asLoveReacterFacade()->reactTo($article, 'Like');
```

#### Pros

- Informative
- Sounds naturally `User as love reacter reacted to article with like`
- Uses same word in:
    - registration method prefix `$user->registerAsLoveReacter()`
    - check for registration method prefix `$user->isRegisteredAsLoveReacter()`

#### Cons

- Might be reserved (see related reading). `as` prefix uses when adapter should be returned.

#### Decision

Candidate №1 (if not reserved in related reading)

#### Related reading

- https://softwareengineering.stackexchange.com/questions/352157/whats-the-difference-between-to-and-as-method-name-prefixes

**Comment by antonkomarev** [comment]
### Name Variant №5

#### Convention

Method should have `get` prefix and `Facade` suffix.

```php
$user->getLoveReacterFacade()->reactTo($article, 'Like');
```

#### Pros

- Very informative

#### Cons

- Very expressive

#### Decision

???

**Comment by antonkomarev** [comment]
### Name Variant №6

#### Convention

Method should have `via` prefix. By way of Reacter.

```php
$user->viaLoveReacter()->reactTo($article, 'Like');
```

#### Pros

- Informative
- Sounds naturally: `User by way of love reacter reacted to article with like`

#### Cons

- Non usual method name prefix

#### Decision

???

**Comment by antonkomarev** [comment]
Implemented in [v7.0 release](https://github.com/cybercog/laravel-love/releases/tag/7.0.0).
